### PR TITLE
[v15] feat: Disable auto-enroll via environment variable

### PR DESCRIPTION
**Pull request by codingllama**
Backport #47679 and #47723 to branch/v15

changelog: Auto-enroll may be locally disabled using the TELEPORT_DEVICE_AUTO_ENROLL_DISABLED=1 environment variable

### CODE REVIEW

**Comment by codingllama** [comment]
This is flaky! Mailed #47723, will merge it here once it lands.

**Review by ryanclark — APPROVED** [review]

**Comment by codingllama** [comment]
PTAL @rosstimothy @ryanclark.

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**
